include/aws/http/connection.h

```diff
@@ -327,8 +340,20 @@ int aws_http2_connection_change_settings(
     struct aws_http_connection *connection,
     const struct aws_http2_setting *settings_array,
     size_t num_settings,
-    void *user_data,
-    aws_http2_on_change_settings_complete_fn *on_completed);
+    aws_http2_on_change_settings_complete_fn *on_completed,
+    void *user_data);
+
+/**
+ * HTTP/2 specific. Send a PING frame with opaque_data as payload.
+ * Opaque_data has to be 8 bytes data or NULL. If opaque_data is NULL, we will send zeroed 8 bytes data.
```

Comment:
trivial: let's start doing doxygen-style. Makes it easy to document the specific details of individual parameters. Some day, if we ever publish docs for aws-c-*** these will become doxygen style

```
/**
 * Send a PING frame (HTTP/2 only).
 * Round-trip-time is calculated when PING ACK is received from peer.
 *
 * @param http2_connection HTTP/2 connection.
 * @param optional_opaque_data Optional payload for PING frame.
 *      Must be NULL, or exactly 8 bytes (AWS_HTTP2_PING_DATA_SIZE).
 *      If NULL, the 8 byte payload will be all zeroes.
 * @param on_completed Optional callback, invoked when PING ACK is received from peer,
 *      or when a connection error prevents the PING ACK from being received.
 *      Callback always fires on the connection's event-loop thread.
 * @param user_data User-data pass to on_completed callback.
 */
```

Reply:
also, let's move `AWS_H2_PING_DATA_SIZE` from h2_frames.h to connection.h and rename it "HTTP2"
it's a useful constant

also, let's rename `connection` here to `http2_connection` to drive home how this is http2 only